src/tss2-fapi/ifapi_eventlog_system.c

```diff
@@ -0,0 +1,766 @@
+/*******************************************************************************
+ * Copyright 2018-2019, Fraunhofer SIT sponsored by Infineon Technologies AG
+ * All rights reserved.
+ *******************************************************************************/
+
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#include <string.h>
+
+#include "ifapi_helpers.h"
+#include "ifapi_eventlog_system.h"
+#include "ifapi_json_serialize.h"
+#include "tpm_json_deserialize.h"
+#include "efi_event.h"
+
+#define LOGMODULE fapi
+#include "util/log.h"
+#include "util/aux_util.h"
+#include "ifapi_macros.h"
+#include "fapi_crypto.h"
+
+static char *tss_const_prefixes[] = { "TPM2_ALG_", "TPM2_", "TPM_", "TPMA_", "POLICY", NULL };
+
+/** Get number from a string.
+ *
+ * A string which represents a number or hex number (prefix 0x) is converted
+ * to an int64 number.
+ *
+ * param[in] token the string representing the number.
+ * param[out] num the converted number.
+ * @retval true if token represents a number
+ * @retval false if token does not represent a number.
+ */
+static bool
+get_number(const char *token, int64_t *num)
+{
+    int itoken = 0;
+    int pos = 0;
+    if (strncmp(token, "0x", 2) == 0) {
+        itoken = 2;
+        sscanf(&token[itoken], "%"PRIx64"%n", num, &pos);
+    } else {
+        sscanf(&token[itoken], "%"PRId64"%n", num, &pos);
```

Comment:
No there is no special reason. if then would still be needed because of the second parameter of sscanf. And I would say the error handling of strtoll is more complicated.

Reply:
> No there is no special reason. if then would still be needed because of the second parameter of sscanf. And I would say the error handling of strtoll is more complicated.

Yeah thats fine, I agree it would just be the if starts with 0x then increment itoken to hold 2 and then call scanf.